### PR TITLE
Upgrading to v2.1.4

### PR DESCRIPTION
**Pull request by metinavc1**
Changed methods for v2.1.4, Added parameter for controlling saved Secrets at beginning of program

### CODE REVIEW

File: acmepos-ui/MainUnit.pas [hunk 1]
```diff
@@ -78,14 +78,17 @@ function FormExists(apForm: TForm): boolean;
...
-  // _eftposAddress := '10.161.106.54';
+  //_eftposAddress := 'emulator-prod.herokuapp.com';
 
   frmMain.edtPosID.Text := _posId;
   frmMain.edtEftposAddress.Text := _eftposAddress;
   if (EncKey <> '') and (HmacKey <> '') then
-    SpiSecrets := ComWrapper.SecretsInit(EncKey, HmacKey);
+  begin
+    SpiSecrets := ComWrapper.SecretsInit(EncKey, HmacKey);
+    Init := True;
```
Comment (gouline) [inline acmepos-ui/MainUnit.pas]: What does `Init` actually represent?

File: acmepos-ui/MainUnit.pas [hunk 2]
```diff
@@ -500,17 +504,25 @@ procedure SecretsChanged(e: SPIClient_TLB.Secrets); stdcall;
...
+  end;
+
+  if (not Init) then
+  begin
+    frmActions.Show;
+
+    if (Spi.CurrentFlow = SpiFlow_Idle) then
+      frmActions.richEdtFlow.Lines.Clear();
+
+    Init := False;
```
Comment (gouline) [inline acmepos-ui/MainUnit.pas]: Isn't `Init` already false if you got into this `then` block?

**Review by gouline — COMMENTED** [review]
Assuming it works, but naming isn't obvious.

File: acmepos-ui/MainUnit.pas (hunk 1 above)
Comment (metinavc1) [reply acmepos-ui/MainUnit.pas]: Init is for Secrets

Comment (gouline) [reply acmepos-ui/MainUnit.pas]: So it's really `SecretsInited`?

Comment (metinavc1) [reply acmepos-ui/MainUnit.pas]: Do you want me to change its name?

File: acmepos-ui/MainUnit.pas (hunk 2 above)
Comment (metinavc1) [reply acmepos-ui/MainUnit.pas]: code is like that..  

```
if (not Init) then
begin
    frmActions.Show;

    if (Spi.CurrentFlow = SpiFlow_Idle) then
      frmActions.richEdtFlow.Lines.Clear();

    Init := False;
end;
```

File: acmepos-ui/MainUnit.pas (hunk 1 above)
Comment (gouline) [reply acmepos-ui/MainUnit.pas]: That would be great.

File: acmepos-ui/MainUnit.pas (hunk 2 above)
Comment (gouline) [reply acmepos-ui/MainUnit.pas]: I can see that. My point is that `Init := False` only gets executed if `not Init` condition evaluates to true (meaning that `Init` is already `False`), right?

Comment (metinavc1) [reply acmepos-ui/MainUnit.pas]: I mixed the place of the code. you're right. there is no need

**Pull request closed without merging**